python/sidecar/references/router.py

```diff
@@ -24,35 +30,39 @@ async def list_references(project_id: str) -> list[Reference]:
     Returns a list of references for the current user
     """
     user_id = "user1"
-    project_path = get_project_path(user_id, project_id)
-    store = storage.JsonStorage(project_path / ".storage" / "references.json")
     try:
-        store.load()
+        store = storage.get_references_json_storage(user_id, project_id)
     except FileNotFoundError:
-        # no references have been ingested yet
         return []
     return store.references
 
 
 @router.post("/{project_id}")
-async def ingest_references(project_id: str, payload: EmptyRequest) -> IngestResponse:
+async def ingest_references(
+    project_id: str, request: IngestibleRequest
+) -> IngestResponse:
     """
-    Ingests references from PDFs in the project uploads directory
+    Creates references from a PDF directory or URL.
     """
     user_id = "user1"
-    project_path = get_project_path(user_id, project_id)
-    uploads_dir = project_path / "uploads"
-    req = IngestRequest(pdf_directory=str(uploads_dir))
-    response = ingest.run_ingest(req)
+
+    if request.type == IngestRequestType.UPLOADS_DIRECTORY:
+        uploads_dir = get_project_uploads_path(user_id, project_id)
+        response = ingest.run_ingest(pdf_directory=uploads_dir)
+
+    elif request.type == IngestRequestType.PDF:
+        reference = create_reference(
+            project_id, metadata=request.metadata, url=request.url
+        )
+        response = IngestResponse(project_name=project_id, references=[reference])
+
     return response
```

Comment:
> Can we add to the to-do list for this pull to return the id of the newly created reference to facilitate removal on the S2 Search dialog (only right after addition ie user made a mistake).

@jvoytek This should be what already happens here. The endpoint returns an `IngestResponse` with a `references` key, which in this case would contain the newly created reference.